playwright/_impl/_set_input_files_helpers.py

```diff
@@ -0,0 +1,80 @@
+import base64
+import os
+import shutil
+from pathlib import Path
+from posixpath import basename
+from typing import TYPE_CHECKING, List, Union
+
+from playwright._impl._helper import Error, async_readfile
+from playwright._impl._writable_stream import WritableStream
+
+if TYPE_CHECKING:  # pragma: no cover
+    from playwright._impl._browser_context import BrowserContext
+
+from playwright._impl._api_structures import FilePayload
+
+SIZE_LIMIT_IN_BYTES = 50 * 1024 * 1024
+
+
+async def convert_input_files(
+    files: Union[str, Path, FilePayload, List[Union[str, Path]], List[FilePayload]],
+    context: "BrowserContext",
+):
+    files = files if isinstance(files, list) else [files]
+
+    has_large_buffer = list(
+        filter(
+            lambda f: not isinstance(f, (str, Path))
+            and len(f.get("buffer", "")) > SIZE_LIMIT_IN_BYTES,
+            files,
+        )
+    )
+    if has_large_buffer:
+        raise Error(
+            "Cannot set buffer larger than 50Mb, please write it to a file and pass its path instead."
+        )
+
+    has_large_file = list(
+        map(
+            lambda f: os.stat(f).st_size,
+            filter(lambda f: isinstance(f, (str, Path)), files),
+        )
+    )
+    if has_large_file:
+        if context._channel._connection.is_remote:
+            streams = []
+            for file in files:
+                stream: WritableStream = await context._channel.send(
+                    "createTempFile", {"name": os.path.basename(file)}
+                )
+                await WritableStream.copy(file, stream)
+                streams.append(stream)
+            return {"streams": streams}
+        return {"localPaths": list(map(lambda f: Path(f).absolute().resolve(), files))}
+
+    return {"files": await normalize_file_payloads(files)}
```

Comment:
I'd probably inline normalize_file_payloads now, since we don't use it elsewhere. e